raster-foundry/raster-foundry
# Fix annotation shapefile export and import

## Overview

This PR fixes annotation shapefile frontend and backend for:
 - import:
    * read and parse projection from `.prj` and re-project to web mercator before storing in db
    * use user specified field names to read and parse properties before storing in db
 - export:
    * re-project from web mercator to wgs84 and set so in the schema

### Checklist

- ~Styleguide updated, if necessary~
- ~Symlinks from new migrations present or corrected for any new migrations~
- ~Any content changes are properly templated using `BUILDCONFIG.APP_NAME`~
- ~Any new SQL strings have tests~

### Demo

Annotation export as shapefile projected to `WGS84`
![video-play-close](https://raw.githubusercontent.com/aaronxsu/MyData/master/gifs/rf-pr-gifs/export.gif)

Annotation import as shapefile with field selection and `.prj` reading under the hood
![video-play-close](https://raw.githubusercontent.com/aaronxsu/MyData/master/gifs/rf-pr-gifs/import.gif)


### Notes

Swagger spec ~and frontend messages upon failures~ will be updated on ~Monday~ Wednesday.

## Testing Instructions

 * Draw some annotations on the frontend and export as shapefile
 * Re-upload these annotations to the same project or a different one. Specify field names and import. Check if it succeeds.
 * Export annotations as geojson or shapefile. Re-project them to another CRS as you like. Re-upload these annotations as zipped shapefile. Check if it works.
 * Upload zipped annotation shapefile without `.shp` or `.prj`. It should fail with alerts notifying users ~(I will add frontend messages/alerts on Monday)~

Closes #3672
